tools/upgrade-deps/tests/cli.test.js

```diff
@@ -2,68 +2,70 @@
 const path = require('path');
 const { spawnSync } = require('child_process');
 const { readFileSync } = require('fs');
-const yarnpkg = require('@yarnpkg/lockfile');
 const rimraf = require('rimraf');
 const {
+	getLockContent,
 	getTmpDirectory,
 	isMinorGt,
 	isMinorLockGT,
 	isMajorGT,
 	isMajorLockGT,
 	isSameVersion,
 	isSameLockVersion,
+	getVersionFromLock,
 } = require('./utils');
 
 const fixturePath = path.join(__dirname, 'fixture', 'basic');
 const bin = path.resolve(__dirname, '..', 'bin', 'cli.js');
 const origin = JSON.parse(readFileSync(path.join(fixturePath, 'package.json')));
-const originLock = yarnpkg.parse(
-	readFileSync(path.join(fixturePath, 'yarn-template.lock')).toString(),
-);
+let originLock;
+
 const originSub = JSON.parse(
 	readFileSync(path.join(fixturePath, 'packages', 'suba', 'package.json')),
 );
 
-describe('talend-upgrade-deps', () => {
+describe.each(['package-lock.json', 'yarn.lock'])('talend-upgrade-deps %s', lock => {
```

Comment:
add a test matrix to guarantie same behavior for each test either yarn lock or package lock